pkg/controller/apicast/apicast_controller.go

```diff
@@ -189,6 +191,41 @@ func (r *ReconcileAPIcast) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileAPIcast) updateStatus(instance *appsv1alpha1.APIcast, reconciler *APIcastLogicReconciler) (reconcile.Result, error) {
+	// Get the Deployment object
+	//   How to do this???
+	// Check if the image field in the Deployment object matches the field in the CR.
+	//   If it does match we do nothing
+	//   If it does not match we update the CR and requeue
+	apicast, err := reconciler.APIcastFromCRContents()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	var apicastDeployment *appsv1.Deployment
+	r.Client().Get(context.TODO(), types.NamespacedName{Name: apicast.DeploymentName, Namespace: apicast.Namespace}, apicastDeployment)
```

Comment:
I understand the way to update status fields is by going to look at the source of truth (in this case the Deployment) instead of focusing on the CR contents.